### DISCUSSION TITLE
Ghostty Website -- Default Option Value

### DISCUSSION THREAD

**Discussion by arithefirst** [ghostty-org/ghostty#3568] [Feature Requests, Ideas]
I was helping some people out in the Discord, and a few people said they wished that the [Option Reference](https://ghostty.org/docs/config/reference) had the default values. I would be willing to PR and add them all in, I just wanted to see if there was a reason we weren't doing this or if it was something @mitchellh specifically didn't want.

**Comment by SenseiMarv** [comment]
I opened a similar discussion, which included a bit more explanation and a solution proposal (https://github.com/ghostty-org/ghostty/discussions/3872). I closed it as duplicate, but here is a copy of it:

When reading the documentation, it is difficult for me to decipher what the default value of a setting is and what values are allowed in Ghostty. While there is often a text description where the default is stated and there is a list showing what values are allowed, e.g. here: https://ghostty.org/docs/config/reference#cursor-style-blink, this is not always clearly stated and makes it difficult to find information quickly by having to parse large chunks of text, e.g. here there is no clear statement of what the default is: https://ghostty.org/docs/config/reference#cursor-style.

I'd like to discuss whether it's possible to introduce a clear short statement of what the default is and what values are allowed/what type the value is. This would probably only affect the [Option Reference](https://ghostty.org/docs/config/reference) page (can be discussed if someone sees a value for adding this to other pages as well).

Examples of what I have in mind:

Example 1:
# [`font-family`](https://ghostty.org/docs/config/reference#font-family)

Value: `string`

The font families to...

---

Example 2:
# [`font-size`](https://ghostty.org/docs/config/reference#font-size)

Value: `number` (Default: `14`) <- Unclear what even is the default here, as there currently is no clear statement.

Font size in points. This value...

---

Example 3:
# [`cursor-style`](https://ghostty.org/docs/config/reference#cursor-style)

Value:
  - `block` (Default)
  - `bar`
  - `underline`
  - `block_hollow`

The style of the cursor. This sets...

**Comment by pluiedev** [comment]
I've implemented this in #4995 — not complete yet because it involves an overhaul of our helpgen logic, but the ideas are there
